src/channel.cr

```diff
@@ -421,11 +421,23 @@ class Channel(T)
   end
 
   private def self.select_impl(ops : Indexable(SelectAction), non_blocking)
+    # ops_locks is a duplicate of ops that can be sorted without disturbing the
+    # index positions of ops
+    if ops.responds_to?(:unstable_sort_by!)
```

Comment:
Wouldn't it be better to just case on the type of `ops`? Like `if ops.is_a?(Array) || ops.is_a?(StaticArray)`